Panic in BitBlt

I have experienced a panic in BitBlt. This is hardly necessary to trigger panics in your package, when you could just return an error.

panic: BitBlt failed

goroutine 12 [running]:
myproject/vendor/github.com/vova616/screenshot.BitBlt(0xffffffff920134ee, 0x0, 0x0, 0x31c, 0x4c, 0xffffffff90011059, 0xbe4, 0x824, 0xcc0020)
myproject/vendor/github.com/vova616/screenshot/screenshot_windows.go:140 +0xed
myproject/vendor/github.com/vova616/screenshot.CaptureRect(0xbe4, 0x824, 0xf00, 0x870, 0x0, 0x0, 0x0)
	myproject/sectionGuard/vendor/github.com/vova616/screenshot/screenshot_windows.go:74 +0x310
main.GetTraybarBackground(0xc04224cf50, 0xc04234b301, 0x0, 0x0)
	myproject/main.go:30 +0xfe

--- a/screenshot_windows.go
+++ b/screenshot_windows.go
@@ -1,7 +1,6 @@
 package screenshot
 
 import (
-	"errors"
 	"fmt"
 	"image"
 	"reflect"
@@ -71,9 +70,8 @@ func CaptureRect(rect image.Rectangle) (*image.RGBA, error) {
 	}
 	defer DeleteObject(obj)
 
-	err := BitBlt(m_hDC, 0, 0, x, y, hDC, rect.Min.X, rect.Min.Y, SRCCOPY)
-	if err != nil {
-		return nil, err
+	if !BitBlt(m_hDC, 0, 0, x, y, hDC, rect.Min.X, rect.Min.Y, SRCCOPY) {
+		return nil, fmt.Errorf("BitBlt failed err:%d.\n", GetLastError())
 	}
 
 	var slice []byte
@@ -127,7 +125,7 @@ func GetLastError() uint32 {
 	return uint32(ret)
 }
 
-func BitBlt(hdcDest HDC, nXDest, nYDest, nWidth, nHeight int, hdcSrc HDC, nXSrc, nYSrc int, dwRop uint) err {
+func BitBlt(hdcDest HDC, nXDest, nYDest, nWidth, nHeight int, hdcSrc HDC, nXSrc, nYSrc int, dwRop uint) bool {
 	ret, _, _ := procBitBlt.Call(
 		uintptr(hdcDest),
 		uintptr(nXDest),
@@ -139,9 +137,7 @@ func BitBlt(hdcDest HDC, nXDest, nYDest, nWidth, nHeight int, hdcSrc HDC, nXSrc,
 		uintptr(nYSrc),
 		uintptr(dwRop))
 
-	if ret == 0 {
-		return errors.New("BitBlt failed")
-	}
+	return ret != 0
 }
 
 func SelectObject(hdc HDC, hgdiobj HGDIOBJ) HGDIOBJ {

--- a/screenshot_windows.go
+++ b/screenshot_windows.go
@@ -1,6 +1,7 @@
 package screenshot
 
 import (
+	"errors"
 	"fmt"
 	"image"
 	"reflect"
@@ -70,8 +71,10 @@ func CaptureRect(rect image.Rectangle) (*image.RGBA, error) {
 	}
 	defer DeleteObject(obj)
 
-	//Note:BitBlt contains bad error handling, we will just assume it works and if it doesn't it will panic :x
-	BitBlt(m_hDC, 0, 0, x, y, hDC, rect.Min.X, rect.Min.Y, SRCCOPY)
+	err := BitBlt(m_hDC, 0, 0, x, y, hDC, rect.Min.X, rect.Min.Y, SRCCOPY)
+	if err != nil {
+		return nil, err
+	}
 
 	var slice []byte
 	hdrp := (*reflect.SliceHeader)(unsafe.Pointer(&slice))
@@ -124,7 +127,7 @@ func GetLastError() uint32 {
 	return uint32(ret)
 }
 
-func BitBlt(hdcDest HDC, nXDest, nYDest, nWidth, nHeight int, hdcSrc HDC, nXSrc, nYSrc int, dwRop uint) {
+func BitBlt(hdcDest HDC, nXDest, nYDest, nWidth, nHeight int, hdcSrc HDC, nXSrc, nYSrc int, dwRop uint) err {
 	ret, _, _ := procBitBlt.Call(
 		uintptr(hdcDest),
 		uintptr(nXDest),
@@ -137,7 +140,7 @@ func BitBlt(hdcDest HDC, nXDest, nYDest, nWidth, nHeight int, hdcSrc HDC, nXSrc,
 		uintptr(dwRop))
 
 	if ret == 0 {
-		panic("BitBlt failed")
+		return errors.New("BitBlt failed")
 	}
 }
 

